lambda/api_update.tf

```diff
@@ -18,7 +18,7 @@ resource "aws_lambda_function" "lambda_api_update_function" {
     }
   }
   lifecycle {
-    ignore_changes = [filename]
+    ignore_changes = [filename, environment]
```

Comment:
Just thinking this through - what happens when `environment` changes for real, e.g. if we add new values? Won't this `ignore_changes` prevent real updates as well as encryption updates?

Reply:
Yep, it will, which I thought would be fine but it will also ignore any variables that are populated from systems manager parameters which isn't good so I had a dig around and found this https://github.com/hashicorp/terraform-provider-aws/issues/960 which just says to use `resource` instead of `data` I've done that and it's encrypting the variables properly but not showing a change every time it's run. I've removed environment from ignore_changes as well.